File: JavaBankSystem/src/util/InputUtil.java
package util;

import java.util.Scanner;

public class InputUtil {
    private static final Scanner scanner = new Scanner(System.in);

    public static String readLine(String msg) {
        System.out.println(msg);
        return scanner.nextLine();
    }

    public static int readInt(String msg) {
        while (true) {
            try {
                System.out.println(msg);
                return Integer.parseInt(scanner.nextLine());
            } catch (NumberFormatException e) {
                System.out.println("Invalid input. Please enter a valid integer.");
            }
        }
    }

    public static double readDouble(String msg) {
        while (true) {
            try {
                System.out.println(msg);
                return Double.parseDouble(scanner.nextLine());
            } catch (NumberFormatException e) {
                System.out.println("Invalid input. Please enter a valid number.");
            }
        }
    }
}

File: CppBankSystem/src/util/InputUtil.cpp
#include "InputUtil.h"
#include <iostream>
#include <limits>

std::string InputUtil::readLine(const std::string& msg) {
    std::cout << msg << std::endl;
    std::string input;
    std::getline(std::cin, input);
    return input;
}

int InputUtil::readInt(const std::string& msg) {
    int val;
    while(true) {
        std::cout << msg << std::endl;
        std::cin >> val;

        if(std::cin.fail()) {
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            std::cout << "Invalid input. Please enter an integer." << std::endl;
        } else {
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            return val;
        }
    }
}

double InputUtil::readDouble(const std::string& msg) {
    double val;
    while(true) {
        std::cout << msg << std::endl;
        std::cin >> val;

        if(std::cin.fail()) {
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            std::cout << "Invalid input. Please enter a number." << std::endl;
        } else {
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            return val;
        }
    }
}